config/src/invariant.rs

```diff
@@ -31,3 +37,83 @@ impl Default for InvariantConfig {
         }
     }
 }
+
+impl InlineConfigParser for InvariantConfig {
+    fn config_key() -> String {
+        INLINE_CONFIG_INVARIANT_KEY.into()
+    }
+
+    fn try_merge(&self, configs: &[String]) -> Result<Option<Self>, InlineConfigParserError> {
+        let overrides: Vec<(String, String)> = Self::overrides(configs);
+
+        if overrides.is_empty() {
+            return Ok(None)
+        }
+
+        let mut conf = *self;
+
+        for pair in overrides {
+            let key = pair.0;
+            let value = pair.1;
+            match key.as_str() {
+                "runs" => conf.runs = parse_u32(key, value)?,
+                "depth" => conf.depth = parse_u32(key, value)?,
+                "fail-on-revert" => conf.fail_on_revert = parse_bool(key, value)?,
+                "call-override" => conf.call_override = parse_bool(key, value)?,
+                _ => Err(InlineConfigParserError::InvalidConfigProperty(key.to_string()))?,
+            }
+        }
+        Ok(Some(conf))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{inline::InlineConfigParser, InvariantConfig};
+
+    #[test]
+    fn unrecognized_property() {
+        let configs = &["forge-config: default.invariant.unknownprop = 200".to_string()];
+        let base_config = InvariantConfig::default();
+        if let Err(e) = base_config.try_merge(configs) {
+            assert_eq!(e.to_string(), "'unknownprop' is an Invalid config property");
+        } else {
+            assert!(false)
+        }
+    }
+
+    #[test]
+    fn successful_merge() {
+        let configs = &["forge-config: default.invariant.runs = 42424242".to_string()];
+        let base_config = InvariantConfig::default();
+        let merged: InvariantConfig = base_config.try_merge(configs).expect("No errors").unwrap();
+        assert_eq!(merged.runs, 42424242);
+    }
+
+    #[test]
+    fn merge_is_none() {
+        let empty_config = &[];
+        let base_config = InvariantConfig::default();
+        let merged = base_config.try_merge(empty_config).expect("No errors");
+        assert!(merged.is_none());
+    }
+
+    #[test]
+    fn merge_is_none_unrelated_property() {
```

Comment:
Hmm I think a clearer name for this test could be:
```suggestion
    fn can_merge_unrelated_properties_into_config() {
```